src/Mvc/Mvc.RazorPages/src/Infrastructure/PageActionDescriptorProvider.cs

```diff
@@ -132,16 +127,28 @@ private static string TransformPageRoute(PageRouteModel model, SelectorModel sel
                 return selectorModel.AttributeRouteModel.Template;
             }
 
-            var segments = pageRouteMetadata.PageRoute.Split('/');
-            for (var i = 0; i < segments.Length; i++)
+            var pageRoute = pageRouteMetadata.PageRoute;
+            if (model.RouteParameterTransformer != null)
             {
-                segments[i] = model.RouteParameterTransformer.TransformOutbound(segments[i]);
+                var segments = pageRouteMetadata.PageRoute.Split('/');
+                for (var i = 0; i < segments.Length; i++)
+                {
+                    segments[i] = model.RouteParameterTransformer.TransformOutbound(segments[i]);
+                }
+
+                pageRoute = string.Join('/', segments);
             }
 
-            var transformedPageRoute = string.Join("/", segments);
+            var template = pageRouteMetadata.RouteTemplate;
+            if (template != null)
+            { 
+                template = AttributeRouteModel.ReplaceTokens(
+                    template, 
+                    descriptor.RouteValues, 
+                    model.RouteParameterTransformer).Replace("//", "/");
```

Comment:
Why is the `Replace` needed?

Reply:
The page value in route dictionary contains "/" prefix. This does not play well with `AttributeRouteModel.ReplaceTokens(...)` because the internals of the method grabs the token, retrieves the value from route dictionary and appends it to transformed string. 

For example given route values:
```
{
    "area" : "admin",
    "page": "/myPage"
}
```
and template `[area]/[page]` the `AttributeRouteModel.ReplaceTokens(...)` method will return `admin//myPage`

Would a comment be enough in this case? Or should I investigate on how to achieve it inside `AttributeRouteModel.ReplaceTokens(...)`, maybe even create a new method just for page tokens because the method has no way of knowing if route values are for controller or page.

Reply:
Might be worthwhile to consider changing the existing `ReplaceTokens` method. I think it would be ok to special case slashes to address this one case.